controller/src/controller/models/domain/rsptag.py

```diff
@@ -130,6 +163,22 @@ class RSPImageTag:
     display_name: str
     """Human-readable display name."""
 
+    date: datetime.datetime | None = None
+    """When the image was created, or as close as we can get to that.
+    We try to derive this from the tag string: For RSP daily or weekly
+    tags (or experimentals in one of those formats), we can calculate
+    this to within a day or a week, which is good enough for display
+    purposes.  Otherwise, we may be able to extract this info from
+    the registry, but even if we can, it may be image upload time
+    rather than creation time.
+    """
```

Comment:
Blank line after the summary sentence in docstrings.

This shouldn't be an optional field. You always set it to something, so you can just declare it as `datetime | None` without the `= None`.That lets you drop the `kw_only` on the child class, although that doesn't really matter for how we use it.